lib/instrumentation/ioredis.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+var stringifySync = require('../util/safe-json').stringifySync
+var shimmer = require('../shimmer')
+var urltils = require('../util/urltils.js')
+var recordRedis = require('../metrics/recorders/redis.js')
+var REDIS = require('../metrics/names').REDIS
+
+
+module.exports = function initialize(agent, redis) {
+  var tracer = agent.tracer
+
+  shimmer.wrapMethod(
+    redis && redis.prototype,
+    'redis.prototype',
+    'sendCommand',
+    function wrapSendCommand(original) {
+      return tracer.wrapFunction(
+        REDIS.OPERATION + 'Unknown',
+        recordRedis,
+        original,
+        wrapper
+      )
+    }
+  )
+
+  function wrapper(segment, args, bind) {
+    segment.name = REDIS.OPERATION + (args[0].name || 'unknown')
+    if (args[0].args && typeof args[0].args !== 'function') {
+      urltils.copyParameters(agent.config,
+        {key: stringifySync(args[0].args[0], 'Unknown')}, segment.parameters)
+    }
+
+    // capture connection info for datastore instance metric
+    segment.port = this.connector.options.port
+    segment.host = this.connector.options.host
+    args[0].promise._then = (function(promise, original){
```

Comment:
We probably do not want to represent these as callback segments (the bind() call is meant specifically for callbacks).  Also, wrapping 'then' here will not work well with chained promises, as far as how they would be displayed in traces.

We are planning to make our instrumentation of promises better, which would take care of this in the future.  For now, I would take this out.  It will make any new segments from inside the 'then' callback siblings of the original call, but at least it will be consistent.